### DISCUSSION TITLE
PHPSpreadsheet can't read XLSX file, which is readable by MsExcel and LibreOfficeCalc. Help, please.

### DISCUSSION THREAD

**Discussion by grig-siren** [PHPOffice/PhpSpreadsheet#2206] [Q&A] [answered]
Hello, people!

I need to read and process some XLSX file, which is created by some unknown third-party software. (See attached file in this message).
When I try to read this file using MsWindows 7 and  Microsoft Excel Starter 2010 - I see 1 sheet and 7 rows in it (normal)
When I try to read this file using Linux and Libre Office Calc 6.0.7 - I see 1 sheet and 7 rows (normal)
When I try to read this file using PHPSpreadsheet 1.18 

    require_once('../vendor/autoload.php');
    use PhpOffice\PhpSpreadsheet\IOFactory;
    $filename = "cases.xlsx";
    echo "filename=" . $filename . "\n";
    if( ( $xls_in = IOFactory::load( $filename ) ) === false )
    {
      echo "error reading file";
      return;
    }
    $xlsSheet_in = $xls_in->getActiveSheet();            // This row throws exception
    $maxRow = $xlsSheet_in->getHighestRow();
    echo "maxRow=". $maxRow . "\n";

I get error message:

    PHP Fatal error:  Uncaught PhpOffice\PhpSpreadsheet\Exception: Your requested sheet index: -1 is out of bounds. The actual 
    number of sheets is 0. in /home/user/svn/webapps/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Spreadsheet.php:686
    Stack trace:
    #0 /home/user/svn/webapps/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Spreadsheet.php(578): 
    PhpOffice\PhpSpreadsheet\Spreadsheet->getSheet()
    #1 /home/user/svn/webapps/creatio_stat/test_xlsx_new.php(11): PhpOffice\PhpSpreadsheet\Spreadsheet->getActiveSheet()
    #2 {main}
      thrown in /home/user/svn/webapps/vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet/Spreadsheet.php on line 686

When I load this file into LibreOfficeCalc, then save it as another file and run the same script on a copy of file - everything is good, and this script prints "maxRow=7"

Why??? This seems to be a bug.

Help, please.

[cases.xlsx](https://github.com/PHPOffice/PhpSpreadsheet/files/6755699/cases.xlsx)

**Comment by oleibman** [comment]
The namespacing used in your file is unexpected by PhpSpreadsheet through release 1.18. However, we have recently implemented the first stage of a project to handle namespaces better. If you read your spreadsheet using master rather than release version, it should no longer throw an exception. Please note that the work is ongoing; you should be able to read all your data, but most of your styling (and possibly other things you would like to see) will be missing until we get to the next phase.

**Reply by grig-siren** [reply]
That's OK for me. I need only cell data. but not styling and others. But, sorry, I am newbie in GitHub. So how can I get master version of PHPSpreadsheet using "composer"?

**Comment by MarkBaker** [comment]
```
composer require phpoffice/phpspreadsheet:dev-master
```

**Reply by grig-siren** [reply]
Something gone wrong. After command

`composer require phpoffice/phpspreadsheet:dev-master --with-all-dependencies`

no changes - exception still is throwing. And files in 'vendor/phpoffice/phpspreadsheet/src/PhpSpreadsheet' directory and below all have modification time a week ago, not now.

**Comment by oleibman** [comment]
Unable to duplicate your problem. When I issue the composer require command above, with or without --with-all-dependencies, and run your script, making sure that require specifies the correct directory, I get the expected output (maxRow=7) and complete without throwing an exception. For the record, the files in my vendor directory all have today's date.

**Reply by grig-siren** [reply, marked as answer]
Thanks. My last problem was not in software, but in configuration files. Now everything is OK

**Reply by oleibman** [reply]
Glad that it's working now. Thank you for the confirmation.

**Comment by iamrgroot** [comment]
@oleibman  Thanks for the work on the namespaces problem. The dev-master version works for now!

Do you have any idea about when the changes for the namespaces will be included in the next release?

I'm using https://github.com/Maatwebsite/Laravel-Excel and I need to manually edit my lock file, otherwise Laravel-Excel won't accept de dev-master branch. Works but kind of hacky 😄

**Reply by MarkBaker** [reply]
It will be in the next release; the only decision is when that release will be made... but it will be sometime this month, probably in the next week or two
